kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -228,7 +227,20 @@ public static String getNamespace(HasMetadata entity) {
    */
   public static String sanitizeName(String name) {
     if(name != null) {
-      return name.replaceAll("[^A-Za-z0-9]+", "-");
+      name = INVALID_LABEL_CHARS_PATTERN.matcher(name).replaceAll("_");
```

Comment:
Do you mean to have both the constant pattern and the line below?

Reply:
D'oh! 🤦🏼